devimint/src/cli.rs

```diff
@@ -180,11 +180,12 @@ pub async fn cleanup_on_exit<T>(
             info!("Received shutdown signal before finishing main process, exiting early");
             Ok(())
         }
-        Ok(Ok(v)) => {
+        Ok(Ok(_v)) => {
             debug!(target: LOG_DEVIMINT, "Main process finished successfully, will wait for shutdown signal");
-            task_group.make_handle().make_shutdown_rx().await.await;
-            debug!(target: LOG_DEVIMINT, "Received shutdown signal, shutting down");
-            drop(v); // execute destructors
+            // TODO: why do we do this?
+            // task_group.make_handle().make_shutdown_rx().await.await;
```

Comment:
Oh. I guess we want to kill all the other stuff that might have gotten spawned.

It hangs for me.

Reply:
Oh, but this is not really correct then. We should send a shutdown and wait on `join_all` kind of thing, not expect someone to shut us down.